### ISSUE TITLE
Non-whole-numbers should be a well-formedness error in tuples

### ISSUE THREAD

**Issue by jpolitz** [brownplt/pyret-lang#778]
```
> t = {1;2}
> t.{1.5}
"<Unknown value: details logged to console>"
```

I think it's getting JS `undefined`.

**Comment by jswrenn** [comment]
Addressed by 1ce89e7e4b2259ee98f045a6b1ad00503f9bdeb2. The change forbids `t.{1.5}` and `t.{~1}`, but not `t.{1.0}` (which _is_ a whole number, but kind of funky). Fractional indices are already forbidden at parse-time (with the appropriate poor error message). 

We should revisit the discussion in 863556d5746d8bdf95ab2a52fb180fc74880c908 (since reverted) about making the `index` parameter `s-tuple-get` an `Expr`, since that would allow the appearance of anything besides an `s-num` in the indexing position to be reported as a well-formedness error.

**Comment by schanzer** [comment]
Fixed on CPO
